konig-schemagen/src/main/java/io/konig/schemagen/sql/SqlTableGenerator.java

```diff
@@ -105,9 +141,16 @@ private SqlColumn column(Shape shape, PropertyConstraint p) {
 				throw new SchemaGeneratorException(message);
 			}
 			FacetedSqlDatatype datatype = datatypeMapper.type(p);
-			boolean nullable = !p.isRequiredSingleValue();
-				
-			return new SqlColumn(predicate.getLocalName(), datatype, keyType, nullable);
+			boolean nullable = !p.isRequiredSingleValue();	
+			String columnName=predicate.getLocalName();
+			if(shape.getUsesAbbreviationScheme()!=null){
+				if(abbreviations!=null){
+					for(String key:abbreviations.keySet()){
+						columnName=columnName.replace(key,abbreviations.get(key));
```

Comment:
This is an extremely inefficient way to perform the replacement.  We should use a [Trie](https://en.wikipedia.org/wiki/Trie) structure to search and replace all terms with the corresponding abbreviation.